pkg/utils/grpc.go

```diff
@@ -2,21 +2,40 @@ package utils
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"net"
+	"sync"
 	"time"
 
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/keepalive"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/k8s/events"
 )
 
 const (
 	maxStreams              = 100000
 	streamKeepAliveDuration = 60 * time.Second
 )
 
+// Server is a construct to run gRPC servers
+type Server struct {
+	name     string
+	cm       *certificate.Manager
+	server   *grpc.Server
+	certName string
+
+	mu     sync.Mutex
+	config tls.Config
+}
+
 // NewGrpc creates a new gRPC server
-func NewGrpc(serverType string, port int, certPem, keyPem, rootCertPem []byte) (*grpc.Server, net.Listener, error) {
+func NewGrpc(serverType string, port int, certName string, cm *certificate.Manager) (*Server, net.Listener, error) {
```

Comment:
This file is meant for a generic `gRPC` server, and not specifically ADS. Passing ADS specific  getters such as `certificate.Manager` indicates it's best to move this code under `pkg/envoy/ads` instead.

In other words, this no longer seems to be a utility function of some kind, like before.

Reply:
I think for any gRPC server we create we would want to support updating the tls config on certificate rotation without restart. Supporting updates on rotation requires access to the `certificate.Manager`. However, I believe at this time a gRPC server is only created for ADS.

Reply:
What I meant to say is that `grpc.go` no longer belongs under `pkg/utils`, seems coupled to ADS requirements, so is better suited to be under `pkg/envoy/ads`.

Reply:
I can move it. If we need another grpc server for any reason we can move it back to `pkg/utils`. Like I stated in my above comment, I think `grpc.go` could live in either location.

Reply:
Is this blocking?

Reply:
I'd prefer to have this addressed with this change, given it's just a `git mv` command with little overhead. But feel free to take a call based on your preference.

Reply:
Updated. Merged with existing ads/grpc.go